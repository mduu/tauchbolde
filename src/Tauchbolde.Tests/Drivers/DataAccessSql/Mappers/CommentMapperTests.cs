using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using Newtonsoft.Json;
using Tauchbolde.Driver.DataAccessSql.DataEntities;
using Tauchbolde.Driver.DataAccessSql.Mappers;
using Xunit;

namespace Tauchbolde.Tests.Drivers.DataAccessSql.Mappers
{
    [UseReporter(typeof(DiffReporter))]
    public class CommentMapperTests
    {
        [Fact]
        public void Map()
        {
            var dataObject = new CommentData
            {
                Id = new Guid("DEBC03E1-FBFC-4895-AFBE-D8E221507C86"),
                AuthorId = new Guid("1C9B6622-2CAA-4FDF-9186-2331EB446616"),
                EventId = new Guid("925CE378-C86E-47EB-B98A-7D5689B5AEA6"),
                Text = "The Text!",
                CreateDate = new DateTime(2019, 12, 13),
                ModifiedDate = new DateTime(2019, 12, 14),
            };

            var domainObject = dataObject.MapTo();
            var newDataObject = domainObject.MapTo();
            
            newDataObject.Should().BeEquivalentTo(dataObject);
            Approvals.VerifyJson(JsonConvert.SerializeObject(
                new
                {
                    dataObject,
                    domainObject,
                    newDataObject
                }));
        }
    }
}