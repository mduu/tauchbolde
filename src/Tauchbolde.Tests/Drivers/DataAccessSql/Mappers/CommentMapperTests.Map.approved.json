{
    "dataObject": {
        "EventId": "925ce378-c86e-47eb-b98a-7d5689b5aea6",
        "Event": null,
        "AuthorId": "1c9b6622-2caa-4fdf-9186-2331eb446616",
        "Author": null,
        "CreateDate": "2019-12-13T00:00:00",
        "ModifiedDate": "2019-12-14T00:00:00",
        "Text": "The Text!",
        "Id": "debc03e1-fbfc-4895-afbe-d8e221507c86"
    },
    "domainObject": {
        "EventId": "925ce378-c86e-47eb-b98a-7d5689b5aea6",
        "AuthorId": "1c9b6622-2caa-4fdf-9186-2331eb446616",
        "Author": null,
        "CreateDate": "2019-12-13T00:00:00",
        "ModifiedDate": "2019-12-14T00:00:00",
        "Text": "The Text!",
        "Id": "debc03e1-fbfc-4895-afbe-d8e221507c86"
    },
    "newDataObject": {
        "EventId": "925ce378-c86e-47eb-b98a-7d5689b5aea6",
        "Event": null,
        "AuthorId": "1c9b6622-2caa-4fdf-9186-2331eb446616",
        "Author": null,
        "CreateDate": "2019-12-13T00:00:00",
        "ModifiedDate": "2019-12-14T00:00:00",
        "Text": "The Text!",
        "Id": "debc03e1-fbfc-4895-afbe-d8e221507c86"
    }
}